### PR TITLE
Update NOTICE.txt

### PR DESCRIPTION
**Pull request by danmaas**
**Summary**
Update NOTICE.txt with latest open-source license info

**Additional Notes**
This follows the recent NOTICE.txt update to `mattermost-server`, PR here: https://github.com/mattermost/mattermost-server/pull/9433

The only difference is that this is an NPM-based project, so the dependency manifest is created by listing the "dependencies" section of all `package.json` files.

- Add new dependencies (react-dom)
- Update homepage and project owner info for all dependencies
- Add SPDX open-source license IDs

Linebreak/whitespace changes in this PR are to match the original LICENSE files supplied with each dependency. This is a one-time change that will make future automated updates smaller.

### CODE REVIEW

File: NOTICE.txt [hunk 1]
```diff
@@ -78,105 +80,82 @@ THE SOFTWARE.
...
 ## electron-context-menu
 
-This product contains a modified portion of 'electron-context-menu', which provides a context menu for Electron apps, by Sindre Sorhus.
+This product contains 'electron-context-menu' by Sindre Sorhus.
+
+Context menu for your Electron app
 
 * HOMEPAGE:
-  * https://github.com/sindresorhus/electron-context-menu
+  * https://github.com/sindresorhus/electron-context-menu#readme
```
Comment (jasonblais) [inline NOTICE.txt]: Curious why `#readme` is attached as part of the URL (similarly in a few other places)

File: NOTICE.txt [hunk 2]
```diff
@@ -78,105 +80,82 @@ THE SOFTWARE.
...
 ## electron-squirrel-startup
 
-This product contains a modified portion of 'electron-squirrel-startup', a default Squirrel.Windows event handler for Electron apps, by mongodb-js.
+This product contains 'electron-squirrel-startup' by Lucas Hrabovsky.
+
+Default Squirrel.Windows event handler for your Electron apps.
 
 * HOMEPAGE:
-  * https://github.com/mongodb-js/electron-squirrel-startup
+  * http://github.com/mongodb-js/electron-squirrel-startup
```
Comment (jasonblais) [inline NOTICE.txt]: Why http vs https?

**Comment by danmaas** [comment]
Hi @jasonblais - the "Homepage" URLs are taken directly from the package registry entries made by their respective authors, e.g.

https://www.npmjs.com/package/electron-context-menu
https://www.npmjs.com/package/electron-squirrel-startup

If you mouse over the "homepage" at right you will see the `#readme` fragment on the first one and lack of HTTPS on the second one.

I considered applying an automatic clean-up to these URLs, but hesitate to over-ride any settings made by the project author.

For now, which of the following would you prefer?

1. respect the exact homepage URL in the author's package registry entry
or
2. apply some automatic clean-ups for homepage URLs? (i.e., remove `#fragment` parts and insist on HTTPS even if the author lists HTTP)

(long-term, as a separate effort, I will be reaching out to dependency authors to help them fix problems with URLs, licenses, and ownership data on their own projects).

**Comment by jasonblais** [comment]
I see. I agree with your assessment that we should respect the exact homepage URL provided by the author. Hence, option 1 sounds good to me.

Thank you @danmaas!

**Comment by danmaas** [comment]
The current update follows option 1 as discussed above.

I've re-run the automation against this repo, including the "modified version of..." logic we discussed [here](https://github.com/mattermost/mattermost-mobile/pull/2163), and there are no new changes.

**Comment by jasonblais** [comment]
Thank you all! Merging

**Comment by yuya-oc** [comment]
> I've re-run the automation against this repo, including the "modified version of..." logic we discussed here, and there are no new changes.

@danmaas Is the automation committed to somewhere? Once it's committed, we can reuse it when we update dependencies.

**Comment by danmaas** [comment]
Hi @yuya-oc - I'm working on a GitHub app that will take care of these NOTICE updates automatically. From now until the app is available, I'll be sending manual updates monthly (or upon request).

**Pull request merged**